### ISSUE TITLE
ethnicity study: `incorrect syntax` error

### ISSUE THREAD

**Issue by CarolineMorton** [opensafely/cohort-extractor#243]
I have just tried to run cohort extract on the server for the [ethnicity study](https://github.com/opensafely/ethnicity-covid-research). I get this error, which I can reproduce by running against the test db locally. 

```
  File "/Library/Frameworks/Python.framework/Versions/3.8/bin/cohortextractor", line 10, in <module>
    sys.exit(main())
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/cohortextractor/cohortextractor.py", line 378, in main
    generate_cohort(options.output_dir, options.expectations_population)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/cohortextractor/cohortextractor.py", line 114, in generate_cohort
    _generate_cohort(output_dir, study_name, suffix, expectations_population)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/cohortextractor/cohortextractor.py", line 123, in _generate_cohort
    study.to_csv(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/cohortextractor/study_definition.py", line 38, in to_csv
    self.backend.to_csv(filename, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/cohortextractor/tpp_backend.py", line 32, in to_csv
    for patient_id in self._to_csv(filename, with_sqlcmd=with_sqlcmd):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/cohortextractor/tpp_backend.py", line 41, in _to_csv
    for line in lines:
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/cohortextractor/mssql_utils.py", line 123, in _mssql_query_to_csv_file
    raise ValueError(final_file.read())
ValueError: Msg 156, Level 15, State 1, Server d9b182a33efe, Line 1370
Incorrect syntax near the keyword 'THEN'.
Msg 102, Level 15, State 1, Server d9b182a33efe, Line 1371
Incorrect syntax near 'ISNULL'.
```

**Comment by evansd** [comment]
@CarolineMorton The issue is that there are mismatched brackets in these expressions here:
https://github.com/opensafely/ethnicity-covid-research/blob/9d7608c4b83a97111c584ebc885808839994ddb7/analysis/study_definition.py#L589-L598

It looks to me like there are a few unnecessary brackets here and it might have helped make the problem more obvious if we removed some of them. Maybe something like this?

```py
{
    "T1DM_EX_OS": """
        insulin_last6mo >= 2 AND t1dm_count >= t2dm_count * 2
    """,
    "T2DM_EX_OS": """
        (insulin_last6mo < 2 AND t2dm_count > 0)
        OR
        (insulin_last6mo >= 2 AND t1dm_count < t2dm_count * 2 AND t2dm_count > 0)
    """,
    "NO_DM": "DEFAULT",
},
```

The wider issue though is that these kind of expressions don't get validated by the dummy data generation routines, which means errors don't get caught until much later in the process (and don't generate very useful error messages either). We've already got a ticket tracking this (#192) but just haven't prioritised it yet.
